nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -607,36 +551,37 @@ def add_electrical_series(
     # channels gains - for RecordingExtractor, these are values to cast traces to uV.
     # For nwb, the conversions (gains) cast the data to Volts.
     # To get traces in Volts we take data*channel_conversion*conversion.
-    channel_conversion = recording.get_channel_gains()
-    channel_offset = recording.get_channel_offsets()
-    unsigned_coercion = channel_offset / channel_conversion
-    if not np.all([x.is_integer() for x in unsigned_coercion]):
-        raise NotImplementedError(
-            "Unable to coerce underlying unsigned data type to signed type, which is currently required for NWB "
-            "Schema v2.2.5! Please specify 'write_scaled=True'."
-        )
-    elif np.any(unsigned_coercion != 0):
-        warnings.warn(
-            "NWB Schema v2.2.5 does not officially support channel offsets. The data will be converted to a signed "
-            "type that does not use offsets."
-        )
-        unsigned_coercion = unsigned_coercion.astype(int)
-    if write_scaled:
-        eseries_kwargs.update(conversion=1e-6)
-    else:
-        if len(np.unique(channel_conversion)) == 1:  # if all gains are equal
-            eseries_kwargs.update(conversion=channel_conversion[0] * 1e-6)
-        else:
+    if recording.has_scaled_traces():
+        channel_conversion = recording.get_channel_gains()
+        channel_offset = recording.get_channel_offsets()
+        unsigned_coercion = channel_offset / channel_conversion
+        if not np.all([x.is_integer() for x in unsigned_coercion]):
+            raise NotImplementedError(
+                "Unable to coerce underlying unsigned data type to signed type, which is currently required for NWB "
+                "Schema v2.2.5! Please specify 'write_scaled=True'."
+            )
+        elif np.any(unsigned_coercion != 0):
+            warnings.warn(
+                "NWB Schema v2.2.5 does not officially support channel offsets. The data will be converted to a signed "
+                "type that does not use offsets."
+            )
+            unsigned_coercion = unsigned_coercion.astype(int)
```

Comment:
@CodyCBakerPhD ATM `unsigned_coercion` is computed but not used...must be propagated to the DataChunkIterator as we did in the dev branch (with the dtype correction)

Reply:
`v0.9.*` returns us to the practice of include offsets as a property of the electrode columns: https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/nwbextractors/nwbextractors.py#L247 thus tests for intan pass without needing the older `unsigned_coercion` part, which means we could remove it. But we're also getting movement on adding offsets to NWB so maybe we should just wait for that most proper of solutions for now